libraries/chain/webassembly/privileged.cpp

```diff
@@ -268,6 +269,12 @@ namespace eosio { namespace chain { namespace webassembly {
          [&]( auto& gprops ) {
               gprops.configuration = config_range.config;
       });
+
+      // only update wasm_allocator pool size when new max_sync_call_depth
+      // is greater than existing one
+      if (config_range.config.max_sync_call_depth > old_max_sync_call_depth) {
```

Comment:
Since the `wasm_alloc_pool` already ignores shrinkage, it may be worthwhile considering avoiding this check here and thus the need to track `old_max_sync_call_depth`, to just keep everything simple. Your choice; just seems unnecessary to check in two places.

Reply:
I wanted to do the check as early as possible. But the simpler, the better.

https://github.com/AntelopeIO/spring/pull/1346/commits/e3007096723155eeb5bc265d7d20dc4ed4ff3e7c